app/src/main/java/com/github/swent/echo/compose/components/searchmenu/SearchMenuFilters.kt

```diff
@@ -205,9 +210,73 @@ fun SortByDisplayer(sortBy: SortBy, sortByCallback: (SortBy) -> Unit) {
     }
 }
 
-/*
- * Composable to get a date from the user.
- * TODO : Implement it, but I spend a bit too much time
- *   on non-working solution so leaving it for next sprint
- */
-@Composable fun DateInputSample(dateOutput: ZonedDateTime) {}
+/** Composable to select the date range for the events */
+@Composable
+fun DateFilter(filters: FiltersContainer, timeSliderCallback: (Float, Float) -> Unit) {
+    var sliderPosition by remember { mutableStateOf(filters.from..filters.to) }
+    Column(modifier = Modifier.width(350.dp)) {
+        // Slider for the date range
+        RangeSlider(
+            value = sliderPosition,
+            steps = 13,
+            onValueChange = { range ->
+                sliderPosition = range
+                timeSliderCallback(sliderPosition.start, sliderPosition.endInclusive)
+            },
+            valueRange = 0f..14f,
```

Comment:
only 14 days range for the search ? that seems a bit low to me

Reply:
That's a good point, especially for associations that can plan events months in advance. But at the same time, I think users that will use this feature will be way more interested in events on the next two weeks than in plannifying their whole calendar. But, the problem is actually that we can't display events if they are planned more than 2 weeks ahead. I can change that and display all events that will take place 2 weeks+ if the user have the right side of the slider full on the right.
It's more about place management, the slider fits perfectly in the search sheet where 2 boxes to select the dates are too big and not as practical. Having a 30 steps slider will also be a pain to select a specific day and be accurate with the finger.

Reply:
that's a good solution ! you just need to display that clearly so thet it's not confusing :)